test/framework/generators/capi.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api/test/framework/exec"
+)
+
+// Generator generates provider components for CAPI
+type ClusterAPI struct {
+	// GitRef defines the git ref. If set, the generator will use kustomize
+	GitRef string
+	// Version defines the release version. If GitRef is not set Version must be set and will not use kustomize
+	Version string
+}
+
+// GetName returns the name of the components being generated.
+func (g *ClusterAPI) GetName() string {
+	return fmt.Sprintf("Cluster API version %s", g.Version)
+}
+
+func (g *ClusterAPI) kustomizePath(path string) string {
+	return fmt.Sprintf("https://github.com/kubernetes-sigs/cluster-api//config/%s?ref=%s", path, g.GitRef)
```

Comment:
This seems error prone, changing the config location would break e2es for every provider 🤔

Reply:
Agreed. This is error prone. Is it worth blocking on?

Reply:
Hmm... I think it is. How we input and apply yaml for CAPI components seem pretty fundamental to the framework, we should try to get this right early if possible. What would be the other options to apply this component?

Reply:
I gave this thought over the weekend. I'm going to push back for a couple of reasons. I don't think this is blocking because

1. It's not required for anyone to use, it's designed as a helper that folks can use or not, if it breaks or doesn't work then it's not strictly going to block anyone.
2. I think we can improve this moving forward and it's not something we have to solve completely before merging this.